src/Nest/CommonAbstractions/Infer/ActionIds/ActionIds.cs

```diff
@@ -34,13 +34,14 @@ public bool Equals(ActionIds other)
 		string IUrlParameter.GetString(IConnectionConfigurationValues settings) => string.Join(",", _actionIds);
 
 		public static implicit operator ActionIds(string actionIds) =>
-			actionIds.IsNullOrEmptyCommaSeparatedList(out var list) ? null : new ActionIds(list);
+			actionIds.IsNullOrEmptyCommaSeparatedList(out var list) ? new ActionIds((string)null) : new ActionIds(list);
 
-		public static implicit operator ActionIds(string[] actionIds) => actionIds.IsEmpty() ? null : new ActionIds(actionIds);
+		public static implicit operator ActionIds(string[] actionIds) =>
+			actionIds.IsEmpty() ? new ActionIds((string)null) : new ActionIds(actionIds);
 
 		public override bool Equals(object obj) => obj is ActionIds other && Equals(other);
 
-		public override int GetHashCode() => _actionIds.GetHashCode();
+		public override int GetHashCode() => _actionIds.OrderBy(s => s).GetHashCode();
```

Comment:
`_actionIds.Sort()` then aggregate a hashcode (unchecked) from each item. `GetHashCode()` here is going to be on the `IOrderedEnumerable<T>` instance and not the contents.